src/test/test_proj_data_info_subsets_spect.cxx

```diff
@@ -0,0 +1,648 @@
+    int num_subsets)
+{
+  cerr << "\tTesting Subset forward projection is consistent with unbalanced subset" << endl;
+
+  if (num_subsets >= template_sino_sptr->get_num_views())
+    {
+      cerr << "Error: Template provided doesn't have enough views to conduct this test with " << num_subsets << " subsets."
+           << std::endl;
+      everything_ok = false;
+    }
+
+  auto full_forward_projection = generate_full_forward_projection(input_image_sptr, template_sino_sptr);
+
+  for (int subset_n = 0; subset_n < num_subsets; ++subset_n)
+    {
+      // subset 0 to subset num_subsets-2 get 1 the view
+      // final subset (num_subsets-1) gets the remainder
+      std::vector<int> subset_views;
+
+      if (subset_n < num_subsets - 1)
+        {
+          subset_views.push_back(subset_n);
+        }
+      else
+        {
+          for (int view = num_subsets - 1; view < full_forward_projection.get_num_views(); view++)
+            {
+              subset_views.push_back(view);
+            }
+        }
+      auto subset_forward_projection_uptr = full_forward_projection.get_subset(subset_views);
+
+      std::ostringstream subset_views_msg;
+      subset_views_msg << '[';
+      for (std::size_t view_idx = 0; view_idx < subset_views.size(); ++view_idx)
+        {
+          if (view_idx != 0)
+            subset_views_msg << ", ";
+          subset_views_msg << subset_views[view_idx];
+        }
+      subset_views_msg << ']';
+      cerr << "\tTesting unbalanced subset " << subset_n << ": views " << subset_views_msg.str() << endl;
+      test_forward_projection_for_one_subset(input_image_sptr, full_forward_projection, *subset_forward_projection_uptr);
+    }
+}
+
+void
+TestProjDataInfoSubsets::test_forward_projection_is_consistent_with_reduced_segment_range(
+    const shared_ptr<const VoxelsOnCartesianGrid<float>>& input_image_sptr,
+    const shared_ptr<const ProjData>& template_sino_sptr,
+    int num_subsets)
+{
+  cerr << "\tTesting Subset forward projection is consistent with reduced segment range" << endl;
+
+  if ((template_sino_sptr->get_min_segment_num() > 0) && (template_sino_sptr->get_max_segment_num() < 1))
+    {
+      cerr << "Error: Template provided doesn't have enough segments to conduct this test with "
+           << template_sino_sptr->get_num_segments() << " segments." << std::endl;
+      everything_ok = false;
+    }
+
+  // First we make a full forward projection with a reduced segment range
+  shared_ptr<ProjDataInfo> reduced_seg_range_pdi_sptr(template_sino_sptr->get_proj_data_info_sptr()->clone());
+
+  // Print the initial segment range
+  std::cerr << "Initial segment range: min_segment_num = " << reduced_seg_range_pdi_sptr->get_min_segment_num()
+            << ", max_segment_num = " << reduced_seg_range_pdi_sptr->get_max_segment_num() << std::endl;
+
+  // Attempt to reduce the segment range
+  reduced_seg_range_pdi_sptr->reduce_segment_range(-1, 1);
```

Comment:
this will be where the assert is raised. Of course,
1) it should never have got here, which means that L453 is wrong. I've created #1627
2) this test doesn't make sense for SPECT